### DISCUSSION TITLE
bind every keydown in each <template> element

### DISCUSSION THREAD

**Discussion by Unknownymous** [alpinejs/alpine#3327] [1. Help]
I just started learning alpine, just wondering if this is possible in alpine.js binding every keydown in each <template> element? What I want to achieve is every keypress(number only) will be put inside the span that is generated via <template>.

here's what I've done so far: [JSFiddle](https://jsfiddle.net/unknownymous/xcndk0uf/9/)

**HTML:**

```
<html>
<head>
    <script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>
</head>
<body x-data="cb" @keypress="clickCapture">
<div id="container">
  <div class="input">
    <template x-for="count in initial.inputLength">
      <span class="box"></span>
    </template>
  </div>  
</div>

   
</body>
</html>
```

**JS:**
```
document.addEventListener('alpine:init', () => {
  Alpine.data('cb', () => ({
    initial: {
      inputLength: 4,
      inputGuess: ""
    },
    clickCapture(e) {
      if (e.key >= 0 && e.key <= 9) {
        this.initial.inputGuess += e.key;
        /* console.log(e.key) */
        console.log(this.initial.inputGuess)

      }

    }

  }))

})
```

**Comment by ekwoka** [comment]
Keypress is deprecated

https://developer.mozilla.org/en-US/docs/Web/API/Element/keypress_event

Try changing this to `keyup` or `keydown` and see if that solves the issue.

You should probably also just refactor to use a single array of characters instead of making your own string and length tracker.
